src/cromshell/utilities/io_utils.py

```diff
@@ -101,6 +103,59 @@ def assert_path_is_not_empty(path: Union[str, Path], description: str) -> None:
             raise EOFError(f"ERROR: {description} is empty: {path}.")
 
 
+def has_nested_dependencies(wdl_path: str or Path) -> bool:
+    """Determine if a WDL has any nested imports."""
+
+    with open(wdl_path, 'r') as rf:
+        for l in rf:
+            if l.startswith('import'):
+                m = re.match(r'import "(.+)"', l)
+
+                if "../" in m.group(1):
+                    return True
+
+    return False
+
+
+def get_flattened_filename(tempdir: tempfile.TemporaryDirectory, wdl_path: str or Path):
+    """Generate the filename to use for flattened WDL files."""
```

Comment:
double-check my description 
```suggestion
    """ Generate hyphen-separated path to use for flattened WDL file path.
    For example: 
    tempdir: /path/2/tempdir/ and wdl_path: /dir/path/2/wdl.wdl
    returns: /path/2/tempdir/dir-path-2-wdl.wdl
    """
```